### PR TITLE
Add offset reset strategy param

### PR DESCRIPTION
**Pull request by tonypizzicato**

### CODE REVIEW

File: consumer.go [hunk 1]
```diff
@@ -17,21 +22,21 @@ type Consumer interface {
...
 }
 
-func NewConsumer(zookeeper string, schemaRepo string) (Consumer, error) {
+func NewConsumer(zookeeper string, schemaRepo string, offsetStrategy string) (Consumer, error) {
 	zConfig := go_kafka_client.NewZookeeperConfig()
 	zConfig.ZookeeperConnect = []string{zookeeper}
 
 	consumerConfig := go_kafka_client.DefaultConsumerConfig()
-	consumerConfig.AutoOffsetReset = go_kafka_client.LargestOffset
+	consumerConfig.AutoOffsetReset = offsetStrategy
```
Comment (jmataya) [inline consumer.go:44]: Can you add some validation to make sure that we're enforcing that `offsetStrategy` is either `OffsetResetSmallest` or `OffsetResetLargest`?

**Review by jmataya — CHANGES_REQUESTED** [review]
This is a good capability to add. I just have one request around validating the contents of the offset parameter.

Also, can you update the instructions in the README about the removal of the `partition` parameter?

File: consumer.go (hunk 1 above)
Comment (tonypizzicato) [reply consumer.go:44]: @jmataya done

**Comment by tonypizzicato** [comment]
what do you mean by `update the instructions in the README about the removal of the partition parameter`? it has no info about this parameter :)

**Comment by jmataya** [comment]
@tonypizzicato Nevermind, I see the change in the README. This looks great!

**Review by jmataya — APPROVED** [review]

**Pull request merged**